src/main/java/uk/gov/beis/els/categories/waterheaters/model/EnergyConsumptionUnit.java

```diff
@@ -1,18 +1,41 @@
 package uk.gov.beis.els.categories.waterheaters.model;
 
-public enum EnergyConsumptionUnit {
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import uk.gov.beis.els.model.Displayable;
+
+public enum EnergyConsumptionUnit implements Displayable {
```

Comment:
In this case I think we want people to provide the enum name in the API rather than the full display name string. So I think we should revert this enum to how it was before, and update the `OpenApiPropertyCustomiser` to handle both standard enums and Displayable ones.
e.g. in `processEnumValues` I think you could change it to something like
```
List<Enum<?>> enumValues = (List<Enum<?>>) Arrays.asList(enumClass.getEnumConstants());

List<String> allowedValues = enumValues.stream()
    .map(e -> {
      if (e instanceof Displayable) {
        return ((Displayable) e).getDisplayName();
      } else {
        return e.name();
      }
    })
    .collect(Collectors.toList());
```
to allow handling any enum (not tested this though)

Reply:
it works, thanks
Done